index.js

```diff
@@ -2172,6 +2205,31 @@ module.delete = function(context, entityName, idOrEntity, callback) {
   }
 }
 
+module.void = function(context,entityName,idOrEntity,callback) {
+  var url;
+  var opts = {};
+  switch(entityName.toLowerCase()) {
+    case 'invoice':
+      url = '/' + entityName.toLowerCase() + '?operation=void';break;
+    case 'salesreceipt':
+    case 'payment':
+    default:
+      url = '/' + entityName.toLowerCase() + '?include=void';
+  }
+  opts.url = url;
+  if (_.isObject(idOrEntity)) {
+    module.request(context, 'post', opts, idOrEntity, callback)
+  } else {
+    module.read(context, entityName, idOrEntity, function(err, entity) {
+      if (err) {
+        callback(err)
+      } else {
+        module.request(context, 'post', opts, entity, callback)
+      }
+    })
+  }
+}
+
```

Comment:
Can't you just call `module.update()` from the new `void*()` methods and pass `opts.qs = { include: 'void' }`?

Reply:
@mcohen01 Yes, that would probably be a simpler solution. Thanks!

Reply:
No we can't do that because these three entity's have a specific URL endpoint. See line 2213-2217.

Reply:
Right, what I mean is, if you add a `void` key to the entity before you call `module.update()` from inside your `void*()` methods, you'll have `include=void` added to the query string. This is handled today on L2179-2181. Basically, users can void SalesReceipts and Payments today, with `updateSalesReceipt()` and `updatePayment()`. Though they can't void invoices because the url param needs to be `operation` not `include`. So, what I'm saying is the `void*()` methods you're adding are nice conveniences, but if think if you just add a check on the `entityName` at L2180, and if it's an invoice, specify `opts.qs = { operation: 'void' }` instead of `opts.qs = { include: 'void' }` and then I think you can just call `update()`.